### ISSUE TITLE
lodash/chaining warning for unchainable methods

### ISSUE THREAD

**Issue by garyking** [wix/eslint-plugin-lodash#184]
Some methods are unchainable. I had a line with `_.compact(_.cloneDeep(test))`, and this plugin told me to chain it instead. Of course, `_(test).cloneDeep().compact()` doesn't work, since `cloneDeep` can't be chained.

Is there an alternative way that this code can be written, or should the rule be updated to exclude all unchainable methods?

At the moment, I have this instead:

```
const cloned = _.cloneDeep(test)
_.compact(cloned)
```

**Comment by ganimomer** [comment]
You could write `_.chain(test).cloneDeep().compact().value()`, which would work.
But I understand this could be confusing.
I think it would be reasonable for the `chaining` rule to accept a parameter to only work for implicit chaining.

**Comment by garyking** [comment]
Thanks, I'll use that chaining method for now.

**Comment by garyking** [comment]
I'm using the new version, with `_.compact(_.cloneDeep(test))` like in the OP, and the `always` setting, yet now it doesn't report an error. I thought it should only not report an error if `implicit` was set?
